Школы с наибольшим числом победителей

Maksonchik88 opened the pull request:
https://informatics.msk.ru/mod/statements/view.php?id=5763&chapterid=111353#1

Maksonchik88 committed:
commit_1

iliadmitriev commented on files/schools_with_the_most_winners.py:
```diff
@@ -0,0 +1,29 @@
+
+scores_list = list(data.values())
+
+max_score = max(list(map(lambda x: x[0], scores_list)))
```
Не совсем понятно зачем ты это делаешь. Что делает эта конструкция? Ищет максимум среди первых элементов списка.

Если это для того, чтобы посчитать глобально по всем школам балл победителя, то это надо по всем спискам в словаре, искать максимум, а потом среди полученного списка максимумов искать общий максимум.

Можно без этого обойтись. Есть решение лучше искать этот максимум в самом первом цикле, где ты считываешь файл.

iliadmitriev reviewed (changes requested):
Глобальный максимум (балл победителя посчитан не верно)

Maksonchik88 committed:
commit_2

iliadmitriev commented on files/schools_with_the_most_winners.py:
```diff
@@ -0,0 +1,28 @@
+    for score in scores:
+        if score == max_score:
+            counter += 1
+    data_count_of_winners[school] = int(counter)
```
Не критично, но у тебя counter и так int.

iliadmitriev commented on files/schools_with_the_most_winners.py:
```diff
@@ -0,0 +1,28 @@
+            counter += 1
+    data_count_of_winners[school] = int(counter)
+
+max_school = (max(data_count_of_winners, key=data_count_of_winners.get))
```
А как работает функция `max` с параметром `key`? Почему в ней указал метод `.get`?

Maksonchik88 replied:
Решил по другому

iliadmitriev reviewed (changes requested):
Опиши почему используешь метод get у словаря и как работает параметр key?

Maksonchik88 committed:
*Инициализировал и заполнил словарь, в котором ключом является количество участниковБ а значиением - список школ.
*Нашел максимум количества победителей
*Отсортировал и вывел школу(школы) в порядке возрастания номера школы с максимальным количеством победителей

iliadmitriev reviewed (approved):
отличное решение!
